File: CounterClassic/app/src/main/java/counter/glung/github/com/counterclassic/MainActivity.java
package counter.glung.github.com.counterclassic;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private int counter = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final TextView counterView = (TextView) findViewById(R.id.counter);
        findViewById(R.id.up).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter++;
                updateCounter(counterView);
            }
        });

        findViewById(R.id.down).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter--;
                updateCounter(counterView);
            }
        });
        updateCounter(counterView);
    }

    private void updateCounter(TextView counterView) {
        counterView.setText(Integer.toString(counter));
    }
}

File: CounterCycle/app/src/main/java/com/glung/github/counter/cycle/MainActivity.kt
package com.glung.github.counter.cycle

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        runCycle(
                { element -> main(this, element) },
                makeActivityDriver(this)
        )
    }
}
